code/game/objects/items/implants/implant_misc.dm

```diff
@@ -114,3 +114,40 @@
 /obj/item/implanter/radio
 	name = "implanter (internal radio)"
 	imp_type = /obj/item/implant/radio
+
+/obj/item/implant/health_monitor
+	name = "health monitor implant"
+	activated = TRUE
+	var/obj/item/device/radio/internal_radio
+	var/triggered_in_crit = FALSE
+
+/obj/item/implant/health_monitor/Initialize()
+	. = ..()
+	internal_radio = new/obj/item/device/radio(src)
+	internal_radio.keyslot = new /obj/item/device/encryptionkey/headset_med
+	internal_radio.subspace_transmission = TRUE
+	internal_radio.listening = FALSE
+	internal_radio.broadcasting = FALSE
+	internal_radio.recalculateChannels()
+
+/obj/item/implant/health_monitor/removed(source, silent = 0, special = 0)
+	if(..())
+		if(iscarbon(source))
+			var/mob/living/carbon/C = source
+			C.adv_health_hud = FALSE
+			to_chat(C, "<span class='notice'>You feel less in-tune with your body.</span>")
+
+/obj/item/implant/health_monitor/implant(mob/living/target, mob/user, silent = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.adv_health_hud = TRUE
+		to_chat(C, "<span class='notice'>You feel more in-tune with your body.</span>")
+	return ..()
+
+/obj/item/implant/health_monitor/on_life(mob/living/carbon/source)
+	if(!triggered_in_crit && source.InCritical())
+		triggered_in_crit = TRUE
+		var/area/location = get_area(src)
+		internal_radio.talk_into(src, "Medical emergency! [source] is in critical condition at [location]!", "Medical", SPAN_ROBOT)
+	else
+		triggered_in_crit = FALSE
```

Comment:
Won't this spam if you're in crit?

Reply:
No. The message is only triggered if triggered_in_crit is also false, and triggered_in_crit is set to true until you exit crit

Reply:
you have it as an AND not an OR so on next on_life it gets set right back to false then repeats over and over and ov...